src/main.rs

```diff
@@ -1,107 +1,102 @@
-                                          .takes_value(true))
-
-                                    .arg( Arg::with_name("ID")
-                                          .help("Specifies the ID and overrides existing notifications with the same ID.")
-                                          .long("id")
-                                          .takes_value(true))
-
-                                    .arg( Arg::with_name("hint")
-                                          .help("Specifies basic extra data to pass. Valid types are int, double, string and byte. Pattern: TYPE:NAME:VALUE")
-                                          .short("h")
-                                          .long("hint")
-                                          .takes_value(true))
-
-                                    .arg( Arg::with_name("category")
-                                          .help("Set a category.")
-                                          .short("c")
-                                          .long("category")
-                                          .takes_value(true))
-
-                                    .arg( Arg::with_name("urgency")
-                                          .help("How urgent is it.")
-                                          .short("u")
-                                          .long("urgency")
-                                          .takes_value(true)
-                                          .possible_values(&urgencies))
-
-                                    .arg( Arg::with_name("debug")
-                                          .help("Also prints notification to stdout")
-                                          .short("d")
-                                          .long("debug"))
-                                    //}}}
-                                    )
-                        .subcommand(SubCommand::with_name("info")
-                                    .about("Shows information about the running notification server")
-                                    )
-                        .subcommand(SubCommand::with_name("server")
-                                    .about("Starts a little notification server for testing")
-                                    )
-
-                        .get_matches();
-
-    if let Some(_matches) = matches.subcommand_matches("server") {
+fn main() -> nResult<()> {
+    let args = Cli::parse();
+
+    match args.command {
         #[cfg(all(unix, not(target_os = "macos")))]
-        {
+        Commands::Server => {
             use notify_rust::server::NotificationServer;
             use std::thread;
             let server = NotificationServer::create();
-            thread::spawn(move || NotificationServer::start(&server, |notification| println!("{:#?}", notification)));
+            thread::spawn(move || {
+                NotificationServer::start(&server, |notification| println!("{:#?}", notification))
+            });
```

Comment:
I'm afraid this will have to go though, there is no server feature anymore in notify-rust 4.x

Reply:
I was wondering. Couldn't tell if I missed it due to feature flags / conditional compile / moved elsewhere, or it was actually gone